tests/test_cmd2.py

```diff
@@ -504,25 +505,60 @@ def test_save_invalid_path(base_app, capsys):
 
 
 def test_output_redirection(base_app):
-    # TODO: Use a temporary directory/file for this file
-    filename = 'out.txt'
-
-    # Verify that writing to a file works
-    run_cmd(base_app, 'help > {}'.format(filename))
-    expected = normalize(BASE_HELP)
-    with open(filename) as f:
-        content = normalize(f.read())
-    assert content == expected
-
-    # Verify that appending to a file also works
-    run_cmd(base_app, 'help history >> {}'.format(filename))
-    expected = normalize(BASE_HELP + '\n' + HELP_HISTORY)
-    with open(filename) as f:
-        content = normalize(f.read())
-    assert content == expected
-
-    # Delete file that was created
-    os.remove(filename)
+    fd, filename = tempfile.mkstemp(prefix='cmd2_test', suffix='.txt')
+    
```

Comment:
@kotfu I've been bitten by this exact issue before!  As the Python documentation clearly states, but is non-intuitive to me at least, "mkstemp() returns a tuple containing an OS-level **handle to an open file** (as would be returned by os.open()) and the absolute pathname of that file":
https://docs.python.org/3.6/library/tempfile.html

So immediately after calling mkstemp, you need to call os.close(fd):

```Python
fd, filename = tempfile.mkstemp(prefix='cmd2_test', suffix='.txt')
os.close(fd)
```

Do the same thing in all of the unit tests you modified please and then things should work great ;-)